lib/measured/conversion.rb

```diff
@@ -0,0 +1,91 @@
+class Measured::Conversion
+  def initialize
+    @base_unit = nil
+    @units = []
+  end
+
+  attr_reader :base_unit, :units
+
+  def set_base(unit_name, aliases: [])
+    add_new_unit(unit_name, aliases: aliases, base: true)
+  end
+
+  def add(unit_name, aliases: [], value:)
+    add_new_unit(unit_name, aliases: aliases, value: value)
+  end
+
+  def unit_names_with_aliases
+    @units.map{|u| u.names}.flatten.sort
+  end
+
+  def unit_names
+    @units.map{|u| u.name}.sort
+  end
+
+  def unit_or_alias?(name)
+    unit_names_with_aliases.include?(name.to_s)
+  end
+
+  def unit?(name)
+    unit_names.include?(name.to_s)
```

Comment:
`Measured::UnitError: Cannot find unit for KG.`

Default to case sensitive unit name matches might be restrictive for no reason.